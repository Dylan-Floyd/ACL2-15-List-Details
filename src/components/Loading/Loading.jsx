import React, { useEffect, useState } from 'react'

export default function Loading() {
  const [dotCount, setDotCount] = useState(0)

  useEffect(() => {
    const intervalRef = setInterval(() => {
      setDotCount(dotCount => dotCount === 3 ? 0 : dotCount + 1)
    }, 250)
    return () => {
      clearInterval(intervalRef)
    }
  }, [])

  return (
    <h2>{'Loading' + '.'.repeat(dotCount)}</h2>
  )
}
