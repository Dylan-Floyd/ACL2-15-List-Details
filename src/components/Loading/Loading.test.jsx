import { render, screen } from '@testing-library/react'
import Loading from './Loading.jsx'

it('should render Loading text and progressively add dots', async () => {
  const { container } = render(<Loading />)

  screen.getByText('Loading')
  await screen.findByText('Loading.')
  await screen.findByText('Loading..')
  await screen.findByText('Loading...')
  await screen.findByText('Loading')

  expect(container).toMatchSnapshot()
})