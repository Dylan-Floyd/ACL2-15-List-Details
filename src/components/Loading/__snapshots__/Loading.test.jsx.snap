// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Loading text and progressively add dots 1`] = `
<div>
  <h2>
    Loading
  </h2>
</div>
`;
